src/mumble/GlobalShortcut_xdp.h

```diff
@@ -0,0 +1,52 @@
+// Copyright 2022 The Mumble Developers. All rights reserved.
+// Use of this source code is governed by a BSD-style license
+// that can be found in the LICENSE file at the root of the
+// Mumble source tree or at <https://www.mumble.info/LICENSE>.
+
+#ifndef MUMBLE_MUMBLE_GLOBALSHORTCUT_XDP_H_
+#define MUMBLE_MUMBLE_GLOBALSHORTCUT_XDP_H_
+
+#include "ConfigDialog.h"
+#include "Global.h"
+#include "GlobalShortcut.h"
+
+#include <QDBusObjectPath>
+
+class OrgFreedesktopPortalGlobalShortcutsInterface;
+
+using XdpShortcut = QPair<QString, QVariantMap>;
+using XdpShortcuts = QList<XdpShortcut>;
+
+class GlobalShortcutXdp : public GlobalShortcutEngine {
+private:
+	Q_OBJECT
+	Q_DISABLE_COPY(GlobalShortcutXdp)
+public:
+	static bool isAvailable();
+
+	void createSession();
+
+	GlobalShortcutXdp();
+	~GlobalShortcutXdp() Q_DECL_OVERRIDE;
+	void run() Q_DECL_OVERRIDE;
+	ButtonInfo buttonInfo(const QVariant &) Q_DECL_OVERRIDE;
+
+	bool canDisable() override;
+	bool canConfigure() override { return true; }
+	void configure() override;
+
+public Q_SLOTS:
```

Comment:
```suggestion
public slots:
```
This is how we usually do this